pkg/types/time.go

```diff
@@ -587,7 +590,7 @@ func GetFracIndex(s string) (index int) {
 	}
 	index = -1
 	for i := end; i >= 0; i-- {
-		if unicode.IsPunct(rune(s[i])) {
+		if s[i] != '+' && s[i] != '-' && unicode.IsPunct(rune(s[i])) {
```

Comment:
We have some functions in `types/time.go` and `types/helper.go`, like `isValidSeparator` and `isPunctuation`. Not sure could we directly use them.

Reply:
It looks like there is no function available.
BTW, I replace `unicode.IsPunct` by `isPunctuation`.

Reply:
Ok, just a question more than suggestion.

Seems `punctuation` in different language has its own meaning. I think `isPunctuation` function is followed `std::ispunct` which may different with `unicode.IsPunct`. Not sure which one is correct.

Reply:
As `isPunctuation` was used consistently in time.go, I have replaced it here. It seems that in time.go, either all function should use `isPunctuation`, or all instances should use `unicode.IsPunct`.

Reply:
I believe replacing it with `isPunctuation` is safe here, as this code snippet was introduced by https://github.com/pingcap/tidb/pull/9933 to handle the case like `2019.01.01 00:00:00`.